dask/base.py

```diff
@@ -1095,7 +1095,9 @@ def _seen_ctx(reset: bool) -> Iterator[dict[int, tuple[int, object]]]:
             # tokenize()
             seen = {}
             tok = _seen.set(seen)
-
+        else:
+            seen = seen.copy()
+            tok = _seen.set(seen)
```

Comment:
So, I'm running into an issue on dask/distributed in the test `test_threadsafe_get`. The tests is launching 30 threads that are concurrently hammering the scheduler with the same computation request over and over again, i.e. the tokens should always be identical and the result should in most likeliness even be reused between all the threads.
However, I'm running into problems and this appears to fix it. Haven't managed to isolate it (already worked with RLocks in various places which didn't help). The problem seems to be a counting issue in the `seen` dictionary but I couldn't reduce it or write a test for it.

I'm slightly concerned about this fix since we call this function quite a lot and copying a non-empty set can be expensive (haven't done any measurements, yet)